File: ReservationDemo/Classes/SqlOperations.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ReservationDemo.Classes
{
    public class SqlOperations
    {
        private static string _connectionString =
            "Data Source=.\\sqlexpress;Initial Catalog=Reservations;Integrated Security=True";

        /// <summary>
        /// Read seats
        /// </summary>
        /// <returns></returns>
        public static List<SeatTable> Read()
        {
            var selectStatement = "SELECT Id, [Row], Number, Available FROM dbo.SeatTable;";
            
            var list = new List<SeatTable>();

            using (var cn = new SqlConnection { ConnectionString = _connectionString })
            {
                using (var cmd = new SqlCommand {Connection = cn, CommandText = selectStatement})
                {
                    cn.Open();
                    var reader = cmd.ExecuteReader();
                    while (reader.Read())
                    {
                        list.Add(new SeatTable()
                        {
                            Id = reader.GetInt32(0), 
                            Row = reader.GetString(1), 
                            Number = reader.GetInt32(2), 
                            Available = reader.GetBoolean(3)
                        });
                    }
                }
            }


            return list;
            
        }
        /// <summary>
        /// Update single seat
        /// </summary>
        /// <param name="identifier">Seat identifier</param>
        /// <param name="available">Seat is available</param>
        public static void UpdateSeat(int identifier, bool available)
        {
            var updateStatement = "UPDATE dbo.SeatTable SET Available = @Available WHERE Id = @Id;";
            
            using (var cn = new SqlConnection {ConnectionString = _connectionString})
            {
                using (var cmd = new SqlCommand {Connection = cn, CommandText = updateStatement })
                {
                    
                    cmd.Parameters.AddWithValue("@Available", available);
                    cmd.Parameters.AddWithValue("@Id", identifier);
                    
                    cn.Open();

                    cmd.ExecuteNonQuery();

                }
            }
        }
    }
}

File: CheckedListBoxDemo/SqlOperations.vb
Imports System.Data.SqlClient

Public Class SqlOperations
    Private Shared ConnectionString As String =
                "Data Source=.\SQLEXPRESS;Initial Catalog=NorthWind2020;Integrated Security=True"

    Public Shared Function Categories() As List(Of Category)
        Dim categoriesList = New List(Of Category)
        Dim selectStatement = "SELECT CategoryID, CategoryName FROM Categories;"

        Using cn As New SqlConnection With {.ConnectionString = ConnectionString}
            Using cmd As New SqlCommand With {.Connection = cn}

                cmd.CommandText = selectStatement

                cn.Open()

                Dim reader = cmd.ExecuteReader()
                While reader.Read()
                    categoriesList.Add(New Category() With {.Id = reader.GetInt32(0), .Name = reader.GetString(1)})
                End While


            End Using

        End Using

        Return categoriesList

    End Function

End Class
Public Class Category
    Public Property Id() As Integer
    Public Property Name() As String

    Public Overrides Function ToString() As String
        Return Name
    End Function
End Class
Public Module Extensions
    <Runtime.CompilerServices.Extension>
    Public Function SetCategory(sender As CheckedListBox, text As String, Optional checkedValue As Boolean = True) As Boolean

        If String.IsNullOrWhiteSpace(text) Then
            Return False
        End If

        Dim result = CType(sender.DataSource, List(Of Category)).
                Select(Function(item, index) New With
                          {
                            Key .Column = item,
                            Key .Index = index
                          }).FirstOrDefault(Function(this) _
                             String.Equals(this.Column.Name, text, StringComparison.OrdinalIgnoreCase))

        If result IsNot Nothing Then
            sender.SetItemChecked(result.Index, checkedValue)
            Return True
        Else
            Return False
        End If
    End Function

End Module